wls-gui-wahllokalsystem/src/composables/tasks/taskListService.ts

```diff
@@ -1,90 +1,76 @@
+    tasks.push(...initWahlvorstandTaskList());
+    tasks.push(...initKonfigurationsparameterTaskList());
+    return tasks;
   }
 
-  const _tasks: Task[] = [
-    {
-      name: "Konfigurationsparameter",
-      onlyForWahlbezirksart: undefined,
-      onlyForWahlen: undefined,
-      onlyForAllWVaehlerverzeichnisse: undefined,
-      callback: () => {
-        return initKonfigurationsparameter(false);
-      },
-    },
-    {
-      name: "Wahlen",
-      onlyForWahlbezirksart: undefined,
-      onlyForWahlen: [
-        WahlWahlartEnum.Obw,
-        WahlWahlartEnum.Bzw,
-        WahlWahlartEnum.Srw,
-      ],
-      onlyForAllWVaehlerverzeichnisse: undefined,
-      callback: () => {
-        return initWahlen(false);
-      },
-    },
-    {
-      name: "Wahlvorstand",
-      onlyForWahlbezirksart: undefined,
-      onlyForWahlen: undefined,
-      onlyForAllWVaehlerverzeichnisse: undefined,
-      callback: () => {
-        return initWahlvorstand(false);
-      },
-    },
-    {
-      name: "UngültigeWahlscheine",
-      onlyForWahlbezirksart: WahlbezirksArtEnum.UWB,
-      onlyForWahlen: [
-        WahlWahlartEnum.Btw,
-        WahlWahlartEnum.Beb,
-        WahlWahlartEnum.Euw,
-        WahlWahlartEnum.Obw,
-        WahlWahlartEnum.Srw,
-        WahlWahlartEnum.Baw,
-      ],
-      onlyForAllWVaehlerverzeichnisse: undefined,
-      callback: () => initUngueltigeWahlscheine(false),
-    },
-    {
-      name: "Kopfdaten",
-      onlyForWahlbezirksart: undefined,
-      onlyForWahlen: undefined,
-      onlyForAllWVaehlerverzeichnisse: undefined,
-      callback: () => {
-        return initKopfdaten();
-      },
-    },
-  ];
+  const taskFactoryData = computed(() => {
+    const extendedWahlMetaData: ExtendedWahlMetaData[] =
+      currentUserWahlMetadata.value.map((wahlMetadata) => {
+        const wahl = getWahlOrUndefinedById(wahlMetadata.wahlID);
+        if (!wahl) {
+          throw new Error(`Wahl not found for wahlID: ${wahlMetadata.wahlID}`);
+        }
+        const extendedWahlMetaData: ExtendedWahlMetaData = {
+          wahlID: wahlMetadata.wahlID,
+          wahlArt: wahl.wahlart,
+          wahlbezirkID: wahlMetadata.wahlbezirkID,
+          wahlName: wahl.name,
+          wahlnummer: wahlMetadata.wahlnummer,
+        };
+        return extendedWahlMetaData;
+      });
+    return {
+      wahlbezirkArt: currentUserWahlbezirksArt.value,
+      extendedWahlMetaData: extendedWahlMetaData,
+    };
+  });
+
+  function initKopfdatenTaskList(): Task[] {
+    return createKopfdatenTasks(taskFactoryData.value);
+  }
+
+  function initUngueltigeWahlscheineTaskList(): Task[] {
+    return createUngueltigeWahlscheineTasks(taskFactoryData.value);
+  }
+
+  function initWahlvorstandTaskList(): Task[] {
+    return createWahlvorstandTasks(taskFactoryData.value);
+  }
+
+  function initKonfigurationsparameterTaskList(): Task[] {
+    return createKonfigurationsparameterTasks(taskFactoryData.value);
+  }
```

Comment:
Ich denke die init-Funktionen sind nicht notwendig. Auch glaube auch nicht dass die extendedWahlMetaData reaktiv sein müssen.

```suggestion
  function _createTaskFactoryData() {
    const extendedWahlMetaData: ExtendedWahlMetaData[] =
      currentUserWahlMetadata.value.map((wahlMetadata) => {
        const wahl = getWahlOrUndefinedById(wahlMetadata.wahlID);
        if (!wahl) {
          throw new Error(`Wahl not found for wahlID: ${wahlMetadata.wahlID}`);
        }
        const extendedWahlMetaData: ExtendedWahlMetaData = {
          wahlID: wahlMetadata.wahlID,
          wahlArt: wahl.wahlart,
          wahlbezirkID: wahlMetadata.wahlbezirkID,
          wahlName: wahl.name,
          wahlnummer: wahlMetadata.wahlnummer,
        };
        return extendedWahlMetaData;
      });
    return {
      wahlbezirkArt: currentUserWahlbezirksArt.value,
      extendedWahlMetaData: extendedWahlMetaData,
    };
  }
```